org.mixedrealitytoolkit.uxcore/Experimental/NonNativeKeyboard/NonNativeKeyTouchAdapter.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+using System.Collections;
+using MixedReality.Toolkit;
+using MixedReality.Toolkit.Input;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace MixedReality.Toolkit.UX.Experimental
+{
+    public class NonNativeKeyTouchAdapter : MonoBehaviour
+    {
+        private const float ColliderMargin = 30.0f;
+        private const float ColliderThickness = 70.0f;
+        private const float ColliderZDelta = 20.0f;
+        private const float ReClickDelayTime = 1.0f;
+        private const float AnimationTime = 0.2f;
+        private const float AnimationMovementDelta = 20.0f;
+
+        private StatefulInteractable interactable;
+        private Graphic image;
+        private float lastClickTime;
+        private bool isInitialized;
+        private Vector3 defaultPosition;
+        private Vector3 animatedPosition;
+        private BoxCollider buttonCollider;
+        private Vector3 buttonColliderDefaultCenter;
+        private Color defaultImageColor;
+
+        private void Awake()
+        {
+            defaultPosition = transform.localPosition;
+            animatedPosition = defaultPosition + new Vector3(0, 0, AnimationMovementDelta);
+        }
+
+        private void OnEnable()
+        {
+            transform.localPosition = defaultPosition;
+            lastClickTime = Time.time;
+            if (isInitialized)
+            {
+                 buttonCollider.center = buttonColliderDefaultCenter;
+            }
+
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            if (isInitialized)
+            {
+                return;
+            }
+            isInitialized = true;
+
+            var rectTransform = GetComponent<RectTransform>();
```

Comment:
Add "Require Component" attribute to this class.

`[RequireComponent(typeof(RectTransform))]`

Reply:
Done